File: src/test/java/kaalka/KaalkaFileTest.java
package kaalka;

import org.junit.jupiter.api.*;
import java.util.*;

public class KaalkaFileTest {
    @Test
    public void testFileChunkEncryptDecrypt() throws Exception {
        String sender = "fileSender";
        String receiver = "fileReceiver";
        String timestamp = KaalkaUtils.utcNow();
        byte[] fileBytes = "This is a test file for Kaalka chunking.".getBytes();
        List<KaalkaEnvelope> chunks = KaalkaFile.encryptFileChunks(fileBytes, sender, receiver, timestamp);
        byte[] decrypted = KaalkaFile.decryptFileChunks(chunks, receiver, timestamp);
        Assertions.assertArrayEquals(fileBytes, decrypted);
    }
}

File: src/test/kotlin/kaalka/KaalkaFileTest.kt
package kaalka

import org.junit.Test
import kotlin.test.assertContentEquals

class KaalkaFileTest {
    @Test
    fun testFileChunkEncryptDecrypt() {
        val sender = "fileSender"
        val receiver = "fileReceiver"
        val timestamp = KaalkaUtils.utcNow()
        val fileBytes = "This is a test file for Kaalka chunking.".toByteArray()
        val chunks = KaalkaFile.encryptFileChunks(fileBytes, sender, receiver, timestamp)
        val decrypted = KaalkaFile.decryptFileChunks(chunks, receiver, timestamp)
        assertContentEquals(fileBytes, decrypted)
    }
}
